katdal/test/test_chunkstore_s3.py

```diff
@@ -46,20 +46,31 @@
 import contextlib
 import io
 import warnings
+import re
+from urllib3.util.retry import Retry
 
 import numpy as np
 from nose import SkipTest
-from nose.tools import assert_raises, assert_equal, timed
+from nose.tools import assert_raises, assert_equal, timed, assert_true, assert_false
 import requests
 import jwt
 
 from katdal.chunkstore_s3 import (S3ChunkStore, _AWSAuth, read_array,
-                                  decode_jwt, InvalidToken)
+                                  decode_jwt, InvalidToken, TruncatedRead,
+                                  _DEFAULT_SERVER_GLITCHES)
 from katdal.chunkstore import StoreUnavailable, ChunkNotFound
 from katdal.test.test_chunkstore import ChunkStoreTestBase
 
 
 BUCKET = 'katdal-unittest'
+# Pick quick but different timeouts and retries for unit tests:
+# The effective connect timeout is 0.2 (initial) + 0.2 (1 retry) = 0.4 seconds
+# The effective read timeout is 0.4 + 0.4 = 0.8 seconds
+# The effective status timeout is 4 * 0.4 + 0.1 * (0 + 2 + 4) = 2.2 seconds
```

Comment:
I was thinking of including the read timeout for the 4 read attempts, but that would only apply to the small header part anyway, because the status error will prevent reading of the main payload. I'll rather leave it out.